sessionctx/variable/varsutil.go

```diff
@@ -591,10 +610,15 @@ func ValidateSetSystemVar(vars *SessionVars, name string, value string, scope Sc
 		TiDBSlowLogThreshold,
 		TiDBQueryLogMaxLen,
 		TiDBEvolvePlanTaskMaxTime:
-		_, err := strconv.ParseInt(value, 10, 64)
+		v, err := strconv.ParseInt(value, 10, 64)
 		if err != nil {
 			return value, ErrWrongValueForVar.GenWithStackByArgs(name)
 		}
+		// These code are for compatibility of future v5.0.0 version.
+		// When rollback from master(future v5.0.0) version, the default value of these variables are -1
+		if v == -1 && name == TiDBProjectionConcurrency {
```

Comment:
why only specially handle `TiDBProjectionConcurrency` in this condition?

Reply:
See https://github.com/pingcap/tidb/issues/15428#issuecomment-655595216

Reply:
sorry, I still can not understand why not need to handle other variables.

Reply:
how about moving this to `if v == -1`?